### PR TITLE
GBM bugfix: matching predictions LightGBM, hummingbird

### PR DESCRIPTION
**Pull request by jppgks**
This PR:

1. Fixes a bug where category variable with 2 classes gave wrong predictions
2. Asserts predictions from LightGBM and converted Hummingbird model match
3. Extends test coverage (incl. for the above cases)
4. Uses cleaner sklearn API for LightGBM

There are quite some changes, the PR is best reviewed commit-by-commit

### CODE REVIEW

File: ludwig/models/gbm.py [hunk 1]
```diff
@@ -28,6 +30,12 @@ def __init__(
         random_seed: int = None,
         **_kwargs,
     ):
+        if len(output_features) > 1:
```
Comment (justinxzhao) [inline ludwig/models/gbm.py:33]: Nice. Thanks for adding these asserts.

File: tests/integration_tests/test_gbm.py [hunk 2]
```diff
@@ -38,21 +35,39 @@ def ray_backend():
...
-    output_features = [text_feature()]
-
+def _train_and_predict_gbm(input_features, output_features, tmpdir, backend_config):
     csv_filename = os.path.join(tmpdir, "training.csv")
     dataset_filename = generate_data(input_features, output_features, csv_filename, num_examples=100)
 
-    config = {MODEL_TYPE: "gbm", "input_features": input_features, "output_features": output_features}
+    config = {
+        MODEL_TYPE: "gbm",
+        "input_features": input_features,
```
Comment (justinxzhao) [inline tests/integration_tests/test_gbm.py:44]: nit: These are also available constants.

**Review by justinxzhao — APPROVED** [review]

**Review by dantreiman — APPROVED** [review]

File: ludwig/models/gbm.py [hunk 3]
```diff
@@ -28,6 +30,12 @@ def __init__(
...
+            raise ValueError("Only single task currently supported")
```
Comment (tgaddair) [inline ludwig/models/gbm.py:34]: For these kinds of errors, I always find it's helpful to include the failing state so the user knows what is wrong. Like:

```
Only single task currently supported, found: {feature_name_1: NUMBER, feature_name_2: CATEGORY}
```

And:

```
Model type GBM only supports number, category, or binary features, found: {feature_name: TEXT}
```
```

File: ludwig/trainers/trainer_lightgbm.py [hunk 4]
```diff
@@ -719,17 +722,17 @@ def set_steps_to_1_or_quit(self, signum, frame):
     def _construct_lgb_params(self) -> Tuple[dict, dict]:
         output_params = {}
         feature = next(iter(self.model.output_features.values()))
-        if feature.type() == CATEGORY:
+        if feature.type() == BINARY or (hasattr(feature, "num_classes") and feature.num_classes == 2):
```
Comment (tgaddair) [inline ludwig/trainers/trainer_lightgbm.py:725]: Why do we need to check the num_classes stuff here when the feature is binary?

**Review by tgaddair — APPROVED** [review]

File: ludwig/trainers/trainer_lightgbm.py (hunk 4 above)
Comment (jppgks) [reply ludwig/trainers/trainer_lightgbm.py:725]: a user can specify a category variable with only two classes, this is to catch that case and explicitly use the binary objective

**Pull request merged**